pshtt/utils.py

```diff
@@ -75,3 +75,28 @@ def configure_logging(debug=False):
         log_level = logging.WARNING
 
     logging.basicConfig(format='%(message)s', level=log_level)
+
+
+# Ensure domains given are not prepended with www/https/http/combination of both.
+def format_domain(domain_name):
+    # Check for www. prepended
+    if domain_name.startswith("www."):
+        return domain_name[4:]
+    # Check for https prepended, account for www. as well
+    if domain_name.startswith("https://"):
+        return format_domain(domain_name[8:])
+
+    # Check for http:// prepended, account for www. as well
+    if domain_name.startswith("http://"):
+        return format_domain(domain_name[7:])
```

Comment:
`http://domain.com` is a valid URL with a valid hostname contained inside it which we want to scan. `http://http://domain.com` isn't a valid URL or hostname, and it's not necessarily the case that the user's intent is to scan `domain.com`. 

In other words, we don't want to get into the business of silently "fixing" invalid data where we don't have information on what the intent of that data was. Instead, we should gracefully surface invalid URLs for correction by the upstream user.